shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/ddl/CreateIndexStatementContext.java

```diff
@@ -49,6 +56,10 @@ public CreateIndexStatementContext(final CreateIndexStatement sqlStatement) {
     
     @Override
     public Collection<IndexSegment> getIndexes() {
-        return null == getSqlStatement().getIndex() ? Collections.emptyList() : Collections.singletonList(getSqlStatement().getIndex());
+        Optional<Integer> startIndex = CreateIndexStatementHandler.getGeneratedIndexStartIndex(getSqlStatement());
+        String generatedLogicIndexName = IndexMetaDataUtil.getGeneratedLogicIndexName(getSqlStatement().getColumns());
+        Collection<IndexSegment> generatedIndexSegments = startIndex.map(index -> Collections.singletonList(new IndexSegment(index, index, 
```

Comment:
if (null != getSqlStatement().getIndex()) {
return
}
Collection<IndexSegment> generatedIndexSegments =xxx;
return?

Reply:
> if (null != getSqlStatement().getIndex()) {
> return
> }
> Collection generatedIndexSegments =xxx;
> return?

@tristaZero Thanks for your suggestion, I have optimized it.